MaxText/layers/linears.py

```diff
@@ -648,6 +690,21 @@ def __call__(self, inputs):
 
     if cfg.megablox:
       max_logging.log("Running MoE megablox implementation.")
+      # This is called only during tracing. This is to invoke creation of quantized tensor.
```

Comment:
Could we wrap line# 693 to 707 into a helper function? i.e. something like process_weights_for_quantization() or others?